packages/compass-collection/src/modules/collection-tab.ts

```diff
@@ -458,6 +563,87 @@ export const analyzeCollectionSchema = (): CollectionThunkAction<
   };
 };
 
+export const generateFakerMappings = (
+  connectionInfo: ConnectionInfo
+): CollectionThunkAction<Promise<void>> => {
+  return async (
+    dispatch,
+    getState,
+    { logger, atlasAiService, preferences }
+  ) => {
+    const { schemaAnalysis, fakerSchemaGeneration, namespace } = getState();
+    if (schemaAnalysis.status !== SCHEMA_ANALYSIS_STATE_COMPLETE) {
+      logger.log.error(
+        mongoLogId(1_001_000_305),
+        'Collection',
+        'Cannot call `generateFakeMappings` unless schema analysis is complete'
+      );
+      return;
+    }
+
+    if (fakerSchemaGeneration.status === 'in-progress') {
+      logger.debug(
+        'Faker mapping generation is already in progress, skipping new generation.'
+      );
+      return;
+    }
+
+    const includeSampleValues = (
+      await preferences.getConfigurableUserPreferences()
+    ).enableGenAISampleDocumentPassing;
+
+    // todo: dedup/abort requests using requestId (CLOUDP-333850)
+    const requestId = new UUID().toString();
```

Comment:
For controlling the abort flow we usually use the thunk arg to keep the reference to the abort controller, see compass-indexes plugin for example

Reply:
didn't see one there but I think this code in [compass-schema-validation](https://github.com/mongodb-js/compass/blob/6585c105daa44c973c01bcdb3c8e67a2359b680c/packages/compass-schema-validation/src/modules/index.ts#L86) that Nataly shared with me works too